c2/c2-protocol/c2-protocol-api/src/main/java/org/apache/nifi/c2/protocol/api/AgentInfo.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Apache NiFi - MiNiFi
+ * Copyright 2014-2018 The Apache Software Foundation
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.c2.protocol.api;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import java.io.Serializable;
+import javax.validation.constraints.NotBlank;
+
+@ApiModel
+public class AgentInfo implements Serializable {
+    private static final long serialVersionUID = -8812289319080770084L;
```

Comment:
Is there a particular reason that all of these model classes implement `Serializable`?  If that is desired, what do you think about setting the initial `serialVersionUID` to `1` for easier tracking and potential incrementing when the structure changes?

Reply:
I'm not sure if `Serializable` _needs_ to be used, I saw an example and did a copy-paste. Does swagger need it? If not then maybe they can be removed

Reply:
Swagger does not need `Serializable`, so if it is not absolutely necessary, it seems cleaner to start without it.

Reply:
I think we may want to enable a caching solution for these POJOs someday, so I'm thinking we could keep them in and set the versions to 1?

Reply:
Keeping `Serializable` and using an initial value of `1` sounds good.